adding criteria operators in approvals (#5300)
harness/developer-hub: docs/continuous-delivery/shared/servicenow-approval-steps.md

Whether the Pipeline/stage stops executing depends on the stage or step [Failure Strategy](/docs/platform/pipelines/define-a-failure-strategy-on-stages-and-steps.md).You can specify criteria using **Conditions** and/or **JEXL Expression**. If you use them in combination, they both must evaluate to `True` for the step to be successful.

> In **Conditions**, you can use the ServiceNow ticket related fields to define approval criteria. Four supported operators are `=`, `!=`, `in` and `not in`. 

Reviewer: ```suggestion
In **Conditions**, you can use the ServiceNow ticket related fields to define approval criteria. Four supported operators are `=`, `!=`, `in`, and `not in`. 
```

For example, 

> - condition for `State` field to be in `Resolved`, `Closed` or `Cancelled` can be specified as:

Reviewer: ```suggestion
- The condition for the `State` field to be in `Resolved`, `Closed`, or `Cancelled` can be specified as:
```

   ![](./static/service-now-approvals-08.png)

> - condition for `State` field to not be in either `New`, `In Progress` or `On Hold` can be specified as:

Reviewer: ```suggestion
- The condition for the `State` field to not be in either `New`, `In Progress`, or `On Hold` can be specified as:
```

   ![](./static/service-now-approvals-09.png)

> - condition for `State` field to be in `Resolved`, `Closed` or `Cancelled` without fixed `Ticket Type` can be specified as:

Reviewer: ```suggestion
- The condition for the `State` field to be in `Resolved`, `Closed`, or `Cancelled` without fixed `Ticket Type` can be specified as:
```

   ![](./static/service-now-approvals-10.png)

:::important

> Multiple conditions with same Field are not allowed. Such usecases can be solved using `in`, `not in` operators or **JEXL Expression**.

Reviewer: ```suggestion
Multiple conditions with the same Jira field are not allowed. Such use cases can be solved using `in`, `not in` operators, or **JEXL Expression**.
```
